cloudify_cli/filters_utils.py

```diff
@@ -49,128 +52,104 @@ def __init__(self, labels_filter_value):
-        """
-        try:
-            raw_rule_key, raw_rule_value = str_filter_rule.split(operator)
-        except ValueError:  # e.g. a=b=c
-            raise cls.format_exception(str_filter_rule)
-
-        rule_key = raw_rule_key.strip()
-        rule_values = cls._get_rule_values(raw_rule_value.strip())
-
-        return rule_key, rule_values
-
     @staticmethod
     def _get_rule_values(raw_rule_value):
+        raw_rule_value = raw_rule_value.replace('\\,', '\x00')
         if raw_rule_value.startswith('[') and raw_rule_value.endswith(']'):
-            return raw_rule_value.strip('[]').split(',')
+            raw_values_list = raw_rule_value.strip('[]').split(',')
+        else:
+            raw_values_list = [raw_rule_value]
 
-        return [raw_rule_value]
+        return [value.replace('\x00', ',') for value in raw_values_list]
 
     def __str__(self, cli_operator):
-        if len(self['values']) == 1:
-            return '{key}{operator}{values}'.format(key=self['key'],
-                                                    operator=cli_operator,
-                                                    values=self['values'][0])
+        values = (self['values'][0] if len(self['values']) == 1 else
+                  '[{0}]'.format(','.join(self['values'])))
 
-        return '{key}{operator}[{values}]'.format(
-            key=self['key'], operator=cli_operator,
-            values=','.join(self['values']))
+        return '"{key}{operator}{values}"'.format(key=self['key'],
+                                                  operator=cli_operator,
+                                                  values=values)
 
 
 class LabelsFilterRule(FilterRule):
-    format_exception = InvalidLabelsFilterRuleFormat
-
     def __init__(self, key, values, operator):
         super(LabelsFilterRule, self).__init__(key, values, operator, 'label')
 
     @classmethod
     def from_string(cls, str_filter_rule):
-        if '!=' in str_filter_rule:
-            key, values = cls._parse_filter_rule(str_filter_rule, '!=')
-            return cls(key, values, OPERATOR_MAPPING['!='])
-
-        elif '=' in str_filter_rule:
-            key, values = cls._parse_filter_rule(str_filter_rule, '=')
-            return cls(key, values, OPERATOR_MAPPING['='])
-
-        elif 'null' in str_filter_rule:
-            match_null = re.match(r'(\S+) is null', str_filter_rule)
-            match_not_null = re.match(r'(\S+) is not null', str_filter_rule)
-            if match_null:
-                key = match_null.group(1).lower()
-                return cls(key, [], OPERATOR_MAPPING['is null'])
-            elif match_not_null:
-                key = match_not_null.group(1).lower()
-                return cls(key, [], OPERATOR_MAPPING['is not null'])
-            else:
-                raise cls.format_exception(str_filter_rule)
-
-        else:
-            raise cls.format_exception(str_filter_rule)
+        match_equal = re.match(r'^([\w\-\.]+)(=)([^\n\t\"]+)$',
```

Comment:
You'll also need to escape = and !, come to think of it

Reply:
As explained on Slack, this is handled by the regexp, so need to escape `=` and `!`.